### PR TITLE
Add branch-level granularity for notebook drag n' drop

### PR DESCRIPTION
**Pull request by omidmogasemi**
Addresses #1298

drag-drop-notebook

### CODE REVIEW

File: polynote-frontend/polynote/ui/component/notebooklist.ts [hunk 1]
```diff
@@ -453,6 +444,43 @@ export class BranchEl extends Disposable {
...
+            this.expanded = true;
+        } else if (evt.type === "drop" || (evt.type === "dragleave" && evt.target === this.dragEnter)) {
+            this.el.classList.remove('highlight-branch');
+            this.expanded = false;
+        }
+
+        // actually handle the file
+        if (evt.type === "drop") {
+            handleFileDrop(evt, this.dispatcher, this.path);
+            this.expanded = true; // keep the folder open so the user can see where their file went
```
Comment (omidmogasemi) [inline polynote-frontend/polynote/ui/component/notebooklist.ts:481]: Should this be the behavior on a successful file drop? Open to hearing opinions :smiley:

**Comment by jeremyrsmith** [comment]
What if we get rid of the borders, and instead put a placeholder item in the spot where you'd be dropping the file?

**Comment by omidmogasemi** [comment]
@jeremyrsmith Agreed! I just edited the original PR with a video of a gray placeholder - let me know what you think!

**Comment by omidmogasemi** [comment]
Yeah, definitely agree - will follow up on that soon (marked it in #1330).

**Pull request merged**